pdns/dnsdist-cache.cc

```diff
@@ -89,15 +89,19 @@ void DNSDistPacketCache::insertLocked(CacheShard& shard, uint32_t key, const DNS
   value = newValue;
 }
 
-void DNSDistPacketCache::insert(uint32_t key, const DNSName& qname, uint16_t qtype, uint16_t qclass, const char* response, uint16_t responseLen, bool tcp, uint8_t rcode)
+void DNSDistPacketCache::insert(uint32_t key, const DNSName& qname, uint16_t qtype, uint16_t qclass, const char* response, uint16_t responseLen, bool tcp, uint8_t rcode, uint32_t tempFailureTTL)
 {
   if (responseLen < sizeof(dnsheader))
     return;
 
   uint32_t minTTL;
 
   if (rcode == RCode::ServFail || rcode == RCode::Refused) {
-    minTTL = d_tempFailureTTL;
+    if (tempFailureTTL != std::numeric_limits<uint32_t>::max()) {
```

Comment:
It might be cleaner to make `tempFailureTTL` a `boost::optional<uint32_t>` instead of relying on a magic value? I know we haven't been consistent in the existing code..